scripts/test_e2e.sh

```diff
@@ -7,12 +7,12 @@ set -xeuo pipefail
 
 REPO_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )/.."
 
-GOMOD_CACHE=$(go env GOPATH)/pkg/mod
+TMPDIR=$(mktemp -d)
+GOMOD_CACHE=$TMPDIR/pkg/mod
 export PATH=${REPO_DIR}/bin:${PATH}
 
 clearGoModCache () {
-  # The sudo is a necessary workaround until go is fixed
-  sudo rm -fr ${GOMOD_CACHE}
+  rm -fr ${GOMOD_CACHE}
```

Comment:
Has the bug where sudo is required fixed yet? I'm still seeing on my machine that I can't remove the mod cache without sudo using go1.11rc1

```
$ go version
go version go1.11rc1 darwin/amd64
$ rm -fr ~/go/pkg/mod/
rm: /Users/carolynvs/go/pkg/mod//github.com/pkg/errors@v0.8.0/example_test.go: Permission denied
```

Reply:
I hadn't checked, this would work around that by placing the mod cache in a user accessible temporary location. Check out ln#11

Reply:
No problem, It was more of a test to start and seemed to work so I ran with it.